data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/MultithreadedProducer.java

```diff
@@ -0,0 +1,101 @@
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.kafka.configuration.KafkaSinkConfig;
+import org.opensearch.dataprepper.plugins.kafka.util.MessageFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * * A Multithreaded helper class which helps to produce the records to multiple topics in an
+ * asynchronous way.
+ */
+@SuppressWarnings("deprecation")
+public class MultithreadedProducer implements Runnable {
+    private KafkaProducer<String, String> plainTextProducer = null;
+    private KafkaProducer<String, JsonNode> jsonProducer = null;
+    private KafkaProducer<String, GenericRecord> avroProducer = null;
+    private static final Logger LOG = LoggerFactory.getLogger(MultithreadedProducer.class);
+    private final AtomicBoolean status = new AtomicBoolean(false);
+    private final KafkaSinkConfig sinkConfig;
+
+    private final Record<Event> record;
+
+    private String schemaType;
+    private Properties producerProperties;
+    private PluginMetrics pluginMetrics;
+
+    public MultithreadedProducer(Properties properties,
+                                 KafkaSinkConfig sinkConfig, Record<Event> record,
+                                 PluginMetrics pluginMetric,
+                                 String schemaType) {
+        this.producerProperties = Objects.requireNonNull(properties);
+        this.sinkConfig = sinkConfig;
+        this.record = record;
+        this.schemaType = schemaType;
+        this.pluginMetrics = pluginMetric;
+        this.jsonProducer = new KafkaProducer<>(producerProperties);
+        this.plainTextProducer = new KafkaProducer<>(producerProperties);
+        this.avroProducer = new KafkaProducer<>(producerProperties);
+
+    }
+
+    @SuppressWarnings({"unchecked"})
+    @Override
+    public void run() {
+        try {
+            MessageFormat schema = MessageFormat.getByMessageFormatByName(schemaType);
+            switch (schema) {
+                case JSON:
+                    new KafkaSinkProducer(jsonProducer, record, sinkConfig, schemaType).produceRecords();
+                    break;
+                case AVRO:
+                    new KafkaSinkProducer(avroProducer, record, sinkConfig, schemaType).produceRecords();
+                    break;
+                case PLAINTEXT:
+                default:
+                    new KafkaSinkProducer(plainTextProducer, record, sinkConfig, schemaType).produceRecords();
+                    break;
+            }
+
+        } catch (Exception exp) {
+            if (exp.getCause() instanceof WakeupException && !status.get()) {
+                LOG.error("Error writing records from the topic...{}", exp.getMessage());
+            }
+        } finally {
+            closeProducers();
+        }
+    }
+
+    private void closeProducers() {
```

Comment:
should be only one procder that needs to be clised.

Reply:
we have addressed in #2950.